### ISSUE TITLE
Huge numbers of "deleted" files with open processes left after Ray Tune run

### ISSUE THREAD

**Issue by jpgard** [ray-project/ray#31556]
### What happened + What you expected to happen

I'm running a series of Ray Tune runs with the Python API (create a `Tuner` object and call `Tuner.fit()`). Each trial takes 100 samples from the hyperparameter search space, and I typically run around a dozen trials for a given experiment (so roughly 1200 iterations). The tuning runs are using `XGBoostTrainer`s and `LightGBMTrainer`s.

The Tuning experiments run as expected, with no errors. However, during/after the runs, there is a **huge** amount of disk space in `/dev/shm` occupied by "deleted" files that, as far as I can tell, are still occupying space because the processes associated with them did not terminate. These files completely fill the directory at `/dev/shm` on my server. It also seems there are severe performance degradations once `/dev/shm` fills up, which typically happens during a training run.

Any suggestions? Is this indeed due to a bug, or is there something that needs to be configured to avoid this?

For others facing this issue, I can work around this by running `kill -9 $(lsof +L1 /dev/shm | grep deleted | awk '{print $2}')`, but this shouldn't be necessary...

Example outputs below:

###########################

```
$ df
Filesystem                             1K-blocks        Used  Available Use% Mounted on
devtmpfs                                    4096           0       4096   0% /dev
tmpfs                                  131976988   126642288    5334700  96% /dev/shm  # <-- was 0% prior to job execution!
tmpfs                                   52790796      716820   52073976   2% /run
/dev/sda2                              209612800    10608692  199004108   6% /
/dev/sdb1                             7810456576    72242124 7738214452   1% /scratch
```

```
$ lsof +L1 /dev/shm
COMMAND       PID   USER   FD   TYPE DEVICE    SIZE/OFF NLINK    NODE NAME
ray::_Inn  795428 jpgard  DEL    REG   0,22                     15312 /dev/shm/plasmacKDpxk
ray::_Inn  795428 jpgard    0u   CHR  136,2         0t0     0       5 /dev/pts/2 (deleted)
ray::_Inn  795428 jpgard   11u   REG   0,22 80532471816     0   15312 /dev/shm/plasmacKDpxk (deleted)
ray::_Inn  800518 jpgard  DEL    REG   0,22                     15312 /dev/shm/plasmacKDpxk
ray::_Inn  800518 jpgard    0u   CHR  136,2         0t0     0       5 /dev/pts/2 (deleted)
ray::_Inn  800518 jpgard   11u   REG   0,22 80532471816     0   15312 /dev/shm/plasmacKDpxk (deleted)
ray::_Inn  803946 jpgard  DEL    REG   0,22                     15312 /dev/shm/plasmacKDpxk
ray::_Inn  803946 jpgard    0u   CHR  136,2         0t0     0       5 /dev/pts/2 (deleted)
ray::_Inn  803946 jpgard   11u   REG   0,22 80532471816     0   15312 /dev/shm/plasmacKDpxk (deleted)
ray::_Inn  811568 jpgard  DEL    REG   0,22                     15312 /dev/shm/plasmacKDpxk
ray::_Inn  811568 jpgard    0u   CHR  136,2         0t0     0       5 /dev/pts/2 (deleted)
ray::_Inn  811568 jpgard   11u   REG   0,22 80532471816     0   15312 /dev/shm/plasmacKDpxk (deleted)

```

### Versions / Dependencies

Python 3.8.15
Ray 2.2.0
Rocky Linux 9.1

### Reproduction script

Sorry, I can't provide a simple reproduction script.

### Issue Severity

High: It blocks me from completing my task.

**Comment by rkooo567** [comment]
I think it is an intended behavior. Ray has an object store that uses /dev/shm, and the processes hold memory once they allocate shared memory. When it is considered deleted, we can still use those memory space for other object store objects. 

Ray is not supposed to use more than /dev/shm than `object_store_memory` which you can see from `ray status`. Once it reaches to the limit, it starts spilling objects to disk, and that may be why you are seeing perf degradationl. 

When you have high /dev/shm usage, what's the output of `ray memory --stats-only`, and what's the output of `ray status`?

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
